play-dl/YouTube/classes/LiveStream.ts

```diff
@@ -172,6 +170,7 @@ export class Stream {
             this.bytes_count += chunk.length
             if(absolute_bytes > (this.per_sec_bytes * 300) && this.per_sec_bytes !== 0){
                 stream.destroy()
+                stream.unpipe(this.stream)
```

Comment:
Remove this line

Add this same line with this.request at setTimeout function.

That would be better.

Reply:
Since this will make un-piping more fast before the actual data is completed.